### ISSUE TITLE
Integrate with SmartTube

### ISSUE THREAD

**Issue by timkgh** [TeamNewPipe/NewPipe#10791]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Integrate with SmartTube. Send video from phone NewPipe app to SmartTube running on Android TV.
SmartTube can already pair with the official YouTube app, maybe it could also pair with NewPipe.

### Why do you want this feature?

Make it easy to flick video from phone to TV

### Additional information

https://github.com/yuliskov/SmartTube?tab=readme-ov-file#casting

**Comment by timkgh** [comment]
> Have you tried using KDE Connect or a similar app?

Not sure I understand how KDE Connect helps pair SmartTube running on Android TV with NewPipe running on my phone. SmartTube can pair with the stock YouTube app on the phone. I'm proposing there should be a way to pair SmartTube with NewPipe also so that one can _cast_ from NP to ST, I'm not looking for alternate solutions to send URLs between devices.

**Comment by timkgh** [comment]
I think there are other ways to do it, using something like BubbleUPnP or PlainUPnp, Yatse/Kore, etc.
I'm looking for a _direct_ integration without adding more apps (and issues).

**Comment by kwijybo** [comment]
@opusforlife2 This attitude is very unhelpful and the suggestion is not related to the issue. please stop.

**Comment by iBicha** [comment]
Hi folks, I'm the author of [Playlet](https://github.com/iBicha/playlet), and I've recently been exploring "the space" for making casting easier. To kinda sum up things
On the phone we have as options
- NewPipe
- ReVanced (which is essentially the official app?)
- Official YouTube app
And on TV we have
- [SmartTube](https://github.com/yuliskov/SmartTube) (AndroidTV)
- [Playlet](https://github.com/iBicha/playlet)
- Official YouTube app

I was exploring how the DIAL protocol works (Link with Wi-Fi option) and essentially I believe NewPipe does not need to implement a special protocol for it to be able to cast to to all 3 TV apps.

In fact I've been prototyping on to cast from the Official YouTube mobile app (and even from the browser) to Playlet.
I have a working "work-in-progress" prototype in Playlet here https://github.com/iBicha/playlet/pull/276, and I have shared my findings and approach in https://github.com/yuliskov/MediaServiceCore/issues/10 because I think the same implementation can work with SmartTube.

So if both SmartTube and Playlet implement the [DIAL spec](https://www.dial-multiscreen.org/) (SmartTube already has the LoungeAPI aka casting) then all NewPipe need to do is to integrate the LoungeAPI to be able to cast to YouTube. Then all the apps listed would be speaking the same protocol and become compatible.

**Comment by exzombie** [comment]
You can use [Open Link With](https://f-droid.org/en/packages/com.tasomaniac.openwith.floss/), which is a "browser" whose sole purpose is redirecting URLs to apps. Make it the default browser on the TV, it makes life better.

That said, this won't solve the issue that the kde connect app needs to be open on the TV; I've tried the LocalSend app, and it has the same issue.  It's just doesn't seem possible to send an URL to the TV in a generic way that would work while NewPipe is open. Switching apps on the TV is not as easy as on a phone, especially for KDE connect which is not added to the home screen and has to be accessed through settings.

**Comment by timkgh** [comment]
This is a lot more complicated than it really needs to be. Quality of life would be improved if NewPipe just supported the TV linking protocol.

**Comment by ilar** [comment]
> This is a lot more complicated than it really needs to be. Quality of life would be improved if NewPipe just supported the TV linking protocol.

Yes, can we please drop the side discussions about KDE Connect and other kludges? They really have no place here, this isn't a tech support chat.

**Comment by veganomy** [comment]
I think the author is talking about "Watch on TV" casting mode. He couldn't frame words around it may be the author's not fond of English or the technical terms.

SmartTube supports cast receiver mode. So YouTube official app can connect with SmartTube and thereby interact wirelessly and listen to player changes such as pause/play, seek, change in the video, etc. It only works when both device and TV are in the same network.

The issue is actually not about SmartTube Integration. It's about "Watch on TV" cast support as a source device, so that TV apps such as official YouTube or SmartTube playback can be controlled from NewPipe.

<img width="719" height="923" alt="Image" src="https://github.com/user-attachments/assets/7bcce33a-a342-42d0-8ae6-fab45ae65632" />

---

<img width="719" height="761" alt="Image" src="https://github.com/user-attachments/assets/fc74af39-de1c-4c55-ad0a-c52d4b62abf9" />

**Comment by thefaboss** [comment]
@veganomy i love that idea :)
